_posts/2017-08-21-Write-Efficient-Deep-Learning-Operators-with-TVM-A-Depthwise-Convolution-Example.md

```diff
@@ -0,0 +1,425 @@
+---
+layout: post
+title:  "Optimize Deep Learning GPU Operators with TVM: A Depthwise Convolution Example"
+date:   2017-08-21
+author: Yuwei Hu
+---
+
+Efficient deep learning operators are at the core of deep learning systems. This blog teaches you how to write efficient GPU operators with the help of [TVM](https://github.com/dmlc/tvm),
```

Comment:
Efficient deep learning operators are at the core of deep learning systems. Usually these operators are hard to get right.  [TVM](https://github.com/dmlc/tvm) makes this much easier for us to do such specific optimizations. This blog teaches you how to write efficient GPU kernels. 

As an example, we use depthwise convolution and demonstrate how can we improve over the already hand optimized CUDA kernel provided by tensorflow. The final version is 2x faster on certain workload than tf-1.2-cudnn